modules/javafx.graphics/src/main/java/com/sun/javafx/css/FixedCapacitySet.java

```diff
@@ -0,0 +1,582 @@
+     * for this collection.
+     *
+     * @param c a collection to check, cannot be {@code null}
+     * @return {@code true} if the given collection contains all elements of this
+     *   collection, otherwise {@code false}
+     */
+    public abstract boolean isSuperSetOf(Collection<?> c);
+
+    /**
+     * Freezes this collection, turning it permanently read-only. After freezing,
+     * any method that would modify the collection will instead throw
+     * {@link UnsupportedOperationException}.
+     *
+     * <p>This can be used to avoid wrapping the collection with an unmodifiable
+     * collection or making a read-only copy.
+     */
+    public final void freeze() {
+        this.frozen = true;
+    }
+
+    /**
+     * Checks if the set is allowed to be mutated, and throws an
+     * {@link UnsupportedOperationException} otherwise.
+     */
+    protected final void ensureNotFrozen() {
+        if (frozen) {
+            throw new UnsupportedOperationException();
+        }
+    }
+
+    /**
+     * A set that can hold 0 or 1 elements.
+     *
+     * @param <T> the element type
+     */
+    private static final class Single<T> extends FixedCapacitySet<T> {
+        private T element;
+
+        @Override
+        public int size() {
+            return element == null ? 0 : 1;
+        }
+
+        @Override
+        public Iterator<T> iterator() {
+            return new Iterator<>() {
+                private boolean hasNext = element != null;
+
+                @Override
+                public boolean hasNext() {
+                    return hasNext;
+                }
+
+                @Override
+                public T next() {
+                    if (!hasNext()) {
+                        throw new NoSuchElementException();
+                    }
+
+                    hasNext = false;
+
+                    return element;
+                }
+            };
+        }
+
+        @Override
+        public boolean isSuperSetOf(Collection<?> c) {
+            return element == null || c.contains(element);
+        }
+
+        @Override
+        public boolean contains(Object o) {
+            return element != null && element.equals(o);
+        }
+
+        @Override
+        public boolean add(T e) {
+            ensureNotFrozen();
+
+            if (contains(Objects.requireNonNull(e, "e"))) {
+                return false;
+            }
+
+            if (element != null) {
+                throw new IllegalStateException("set is full");
+            }
+
+            element = e;
+
+            return true;
+        }
+
+        @Override
+        public boolean addAll(Collection<? extends T> c) {
+            boolean modified = false;
+
+            for (T element : c) {
+                modified |= add(element);
+            }
+
+            return modified;
+        }
+
+        @Override
+        public int hashCode() {
+            return element == null ? 0 : element.hashCode();
```

Comment:
We can't, `hashCode` for `Set`s are specified:

> The hash code of a set is defined to be the sum of the hash codes of the elements in the set, where the hash code of a {@code null} element is defined to be zero

This is done so that using a set as a key would hash to the same value regardless of what type of set it is.